runtime/vm/JVMImage.cpp

```diff
@@ -74,30 +125,69 @@ JVMImage::freeSubAllocatedMemory(void* address)
 }
 
 void
-JVMImage::reallocateImageMemory(UDATA size)
+JVMImage::readImageFromFile()
 {
-	return;
+    OMRPORT_ACCESS_FROM_OMRPORT(getPortLibrary());
+
+    char imageBuffer[_imageSize];
+    memset(imageBuffer, 0, sizeof(imageBuffer));
+
+    intptr_t fileDescriptor = omrfile_open(_dumpFileName, EsOpenRead, 0666);
+    if (fileDescriptor == -1) {
+        // Failure to open file
+    }
+
+    intptr_t bytesRead = omrfile_read(fileDescriptor, imageBuffer, sizeof(imageBuffer));
+    if ((bytesRead == 0) || (bytesRead == -1)) {
+        // Failure to read the image
+    }
+
+    if (omrfile_close(fileDescriptor) != 0) {
+        // Failure to close
+    }
+
+    // Finally load the JVMImage using the data we read from the image
 }
 
-JVMImage *
-JVMImage::getJVMImage() {
-	return NULL;
+void
+JVMImage::storeImageInFile()
+{
+    OMRPORT_ACCESS_FROM_OMRPORT(getPortLibrary());
+
+    intptr_t fileDescriptor = omrfile_open(_dumpFileName, EsOpenCreate | EsOpenWrite | EsOpenTruncate, 0666);
+    if (fileDescriptor == -1) {
+        // Failure to open file
+    }
+
+    if (omrfile_write(fileDescriptor, (void*)_heap, _size) != (intptr_t)_size) {
```

Comment:
possible null on _heap, but unlikely if called correctly. You could put a fast fail above these functions to see if image is allocated

Reply:
close if you think its not needed